### PR TITLE
Extensions

### PR DESCRIPTION
**Pull request by zenhack**
Superceeds #430.

### CODE REVIEW

**Comment by gsilvis** [comment]
This looks good right now.  I still would like to see it used in practice, to check that it's really the right API for this.

**Comment by gsilvis** [comment]
Just noticed/was told that this is not aimed at master.  +1, then

File: haas/test_common.py [hunk 1]
```diff
@@ -45,20 +45,36 @@ def releaseDB(db):
...
+    ``config_dict`` should be a dictionary mapping section names (strings)
+    to dictionaries mapping option names within a section (again, strings)
+    to their values.
+    """
+    config_clear()
+    for section in config_dict.keys():
+        cfg.add_section(section)
+        for option in config_dict[section].keys():
+            cfg.set(section, option, config_dict[section][option])
+
```
Comment (gsilvis) [inline haas/test_common.py]: This function seems almost the same as ``config_merge`` in #437 .  Same purpose, almost the same implementation.  Should the code be shared?

Comment (zenhack) [inline haas/test_common.py]: Ack, it totally should be shared. I probably wrote this one first and forgot about it --- just dealt with the merge on another branch, now I know why it was so big...

This one isn't going into master anyway; if we decide what to do with #437, I can merge it back into this.

**Comment by gsilvis** [comment]
I'm assuming you don't want to merge this until we get #437 merged?

**Comment by zenhack** [comment]
@gsilvis probably makes sense, yeah.

**Comment by zenhack** [comment]
subsumed by #455; closing

**Pull request closed without merging**